bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/VigiCruesHandlerFactory.java

```diff
@@ -64,10 +72,13 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
     protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
-        if (thingTypeUID.equals(THING_TYPE_VIGI_CRUES)) {
-            return new VigiCruesHandler(thing, timeZoneProvider, gson);
-        }
+        return supportsThingType(thingTypeUID) ? new VigiCruesHandler(thing, locationProvider, apiHandler) : null;
+    }
 
-        return null;
+    @Override
+    protected void removeHandler(ThingHandler thingHandler) {
+        if (serviceReg != null) {
+            serviceReg.unregister();
+        }
     }
```

Comment:
You don't even check if the handler being removed is even related to the service you are unregistering?
Typically you keep track of a map of handler instances to registered services.